src/bsoncxx/oid.hpp

```diff
@@ -37,27 +37,10 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
 ///
 class BSONCXX_API oid {
    public:
-    struct init_tag_t {};
-
-    // TODO(MSVC): Ideally this would be constexpr, but VS2015U1 can't
-    // handle it.
-    //
-    // See https://connect.microsoft.com/VisualStudio/feedback/details/2092790
-    //
-    static const init_tag_t init_tag;
-
-    ///
-    /// Constructs an uninitialized oid.
-    ///
-    oid();
-
     ///
     /// Constructs an oid and initializes it to a newly generated ObjectId.
     ///
-    /// @param tag
-    ///   A bsoncxx::oid::init_tag used to dispatch this overload.
-    ///
-    explicit oid(init_tag_t tag);
+    oid();
 
     ///
     /// Constructs an oid initializes it to the contents of the provided buffer.
```

Comment:
Also interesting to consider: since we have already called bson_oid_is_valid, and have not proceeded, can we assume that it is OK to pass non-null terminated? Again, @ajdavis, thoughts?